### ISSUE TITLE
/updates/?packages queries timing out

### ISSUE THREAD

**Issue by ralphbean** [fedora-infra/bodhi#271]
```
http get https://bodhi.fedoraproject.org/updates/\?packages\=gnome-shell
```

The query is causing a ton of CPU churn it seems, and never returns.

This is a re-post of fedora-infra/fedmsg_meta_fedora_infrastructure#314, which was filed there accidentally.

**Comment by ralphbean** [comment]
This [double join](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/services/updates.py#L181-L184) is probably to blame?

**Comment by lmacken** [comment]
This issue has gotten a little better, as the query takes about ~20 seconds and eventually returns... but there is still a lot of room for improvement.

**Comment by mattiaverga** [comment]
fixed in #4048
